### PR TITLE
Aos_2: doc fixes

### LINKED ISSUE #7693: Arrangement_on_surface_2: unknown group in documentation
Adding a warning to doxygen gives me the warning:
```
.../Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_2.h:200: warning: Found non-existing group 'insert' for the command '@ingroup', ignoring command
.../Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_2.h:200: warning: Found non-existing group 'The' for the command '@ingroup', ignoring command
.../Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_2.h:200: warning: Found non-existing group 'function' for the command '@ingroup', ignoring command
.../Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_2.h:200: warning: Found non-existing group 'tt' for the command '@ingroup', ignoring command
.../Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_2.h:200: warning: Found non-existing group 'insert' for the command '@ingroup', ignoring command
.../Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_2.h:200: warning: Found non-existing group 'tt' for the command '@ingroup', ignoring command
.../Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_on_surface_2.h:1055: warning: Found non-existing group 'insert' for the command '@ingroup', ignoring command
```

Regarding the first set of warnings I see on line 163ff of Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_2.h
```
/*! \ingroup PkgArrangementOnSurface2Insert insert The function `%insert`
 * inserts one or more curves or \f$ x\f$-monotone curves into a given
 * arrangement, where no restrictions are imposed on the inserted curves. If an
```
and as `\ingroup` can have multiple labels the warning results from this unexpected set of labels.

Regarding the last warning a similar problem exists as at line 1017ff of Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_on_surface_2.h I see:
```
/*! \ingroup PkgArrangementOnSurface2Insert insert
 * The function `%insert` inserts one or more curves or \f$ x\f$-monotone
 * curves into a given arrangement, where no restrictions are imposed on the
```
- what will be the correct formulation?

### PR DESCRIPTION
**Pull request by efifogel**
## Summary of Changes

Fixes documentation issues in the Aos_2 package

## Release Management

* Affected package(s):  Arrangement_on_surface_2
* Issue(s) solved (if any): fix #7693 and #7659
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature)
* License and copyright ownership: TAU

### CODE REVIEW

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+   * \param x2 The second algebraic value.
+   * \pre The two values are not equal.
+   * \return A rational number that lies in the open interval (x1, x2).
+   */
+  Rational rational_in_interval(const Algebraic& x1, const Algebraic& x2) const;
+
+  /*! Obtain a range of double-precision floats that contains the given
+   * algebraic number.
+   * \param x The given number.
+   * \return A pair <x_lo, x_hi> that contain x.
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:68]: Some backticks?

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+   */
+  std::pair<double, double> double_interval(const Algebraic& x) const;
+
+  /*! Convert a sequence of rational coefficients to an equivalent sequence
+   * of integer coefficients. If the input coefficients are q(1), ..., q(k),
+   * where q(i) = n(i)/d(i) then the output coefficients will be of the
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:74]: "formulas" in backticks?

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+   * form:
+   *               n(i) * lcm {d(1), ... , d(k)}
+   *       a(i) = -------------------------------
+   *               d(i) * gcd {n(1), ... , n(k)}
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:78]: total between triple backtics?

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+   *
+   * \param q_begin The begin iterator of the rational sequence.
+   * \param q_end The past-the-end iterator of the rational sequence.
+   * \param zoi An output iterator for the integer coefficients.
+   * \return A past-the-end iterator for the output container.
+   * \pre The value type of q_begin and q_end is `Rational`, and
+   *      the value type of zoi is `Integer`.
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:85]: Backticks here should be  single quotes?
zoi, q_begin and q_end between backtics?

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+  /*! Compute the roots of a quadratic equations \f$a*x^2+ b*x + c = 0\f$
+   * with integer coefficients.
+   * \param a The coefficient of \f$x^2\f$
+   * \param b The coefficient of \f$x\f$
+   * \param c The free term.
+   * \param oi An output iterator for the real-valued solutions of the
+   *           quadratic equation.
+   * \return A past-the-end iterator for the output container.
+   * \pre The value type of oi is Algebraic.
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:107]: oi in backticks?

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+   */
+  template <typename InputIterator, typename OutputIterator>
+  OutputIterator convert_coefficients(InputIterator q_begin,
+                                      InputIterator q_end,
+                                      OutputIterator zoi) const;
+
+  /*! Compute the square root of an algebraic number.
+   * \param x The number.
+   * \return The square root of x.
+   * \pre x is non-negative.
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:95]: x in backticks?

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+                                  unsigned int degree) const;
+
+  /*! Construct a polynomial with integer coefficients given rational
+   * coefficients.
+   * \param coeffs The coefficients of the input polynomial.
+   * \param degree The degree of the input polynomial.
+   * \param poly Output: The resulting polynomial with integer coefficients.
+   * \param poly_denom Output: The denominator for the polynomial.
+   * \return Whether this polynomial is non-zero (false if the polynomial is
+   *         zero).
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:128]: false and zero in backticks?

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+   */
+  bool construct_polynomial(const Rational *coeffs,
+                            unsigned int degree,
+                            Polynomial& poly, Integer& poly_denom) const;
+
+  /*! Construct two polynomials with integer coefficients such that P(x)/Q(x)
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h:134]: Stopped reviewing for backticks ...

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/Arrangement_on_surface_2.txt [hunk 9]
```diff
@@ -5651,10 +5651,24 @@ must model the basic concept `ArrangementBasicTopologyTraits`. A model
...
+`Arr_unb_planar_topology_traits_2`, and
+`Arr_spherical_topology_traits_2`. The first two are internally used
+to define any instance of the class template
+`Arrangement_2<GeometryTraits_2, Dcel>`. In particular, an instance
+`Arrangement_2<Geometry_traits_2, Dcel_>` is derived from the instance
+`Arrangement_on_surface_2<Geometry_traits_2,Topology_traits>`, where
+the `Topology_traits` type is selected based on the provided geometry
+traits `Geometry_traits_2`, or more precisely, on the boundary
+conditions defined by the geometry traits. If all sides of the
+boundary of the parameter space are cosed, the instance
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/Arrangement_on_surface_2.txt:5666]: cosed -> closed

**Comment by efifogel** [comment]
All good points.
I believe that I addressed them all.
I also seized the opportunity and applied several other fixes mainly
related to output iterators.

NOTE: I wasn't able to fix the following perhaps due to a bug in doxygen.
It produces the following erroneous text (look for whereTypeis):
PreconditionDereferencing oi must yield an object of type std::pair<
Arrangement_on_surface_2::Vertex_const_handle,
std::pair<std::optional<Type,std::optional<Type>>>, whereTypeis
std::variant<Arrangement_on_surface_2::Vertex_const_handle,
Arrangement_on_surface_2::Halfedge_const_handle,
Arrangement_on_surface_2::Face_const_handle>`.
in the manual of decompose().
The source is
Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arr_vertical_decomposition_2.h
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by albert-github** [comment]
- which version of doxygen did you test with?
- what is exactly the problem (it is of course hard to describe over email),

**Comment by efifogel** [comment]
1. doxygen --version
1.10.0 (7962329e20472152221c3e75ddafd81b9e8fca99)

2. It removes the spaces and results with: whereTypeis
while it should be: where Type is
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by albert-github** [comment]
Do I see it correctly in [Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arr_vertical_decomposition_2.h](https://github.com/CGAL/cgal/pull/7713/files#diff-aa8ff8c1b909e7f9f7e238637db2ba8f3ae181cf7b11b8fae56a4abccdf34491) around line 43:
```
 * \pre Dereferencing `oi` must yield an object of type
 * `std::pair<Arrangement_on_surface_2::Vertex_const_handle,
 *            std::pair<std::optional<Type,std::optional<Type>>>,
 * where `Type` is
 ```
the `<Type>>>,` shouldn't this be ``<Type>>>`,``?
(so it looks to me that a backtick is missing)

**Comment by efifogel** [comment]
Indeed. Fixed. Thanks!!!
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

File: Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_on_surface_2.h [hunk 10]
```diff
@@ -1295,38 +1294,45 @@ bool remove_vertex
...
  *
  * \pre If provided, `pl` must be attached to the given arrangement `arr`.
+ * \pre The instantiated `GeometryTraits` class must model the
+ *      `ArrangementXMonotoneTraits_2` concept.
+ * \pre The point-location object `pl`, must model the
+ *      `ArrangementPointLocation_2` concept.
+ * \pre Dereferencing `oi` must yield a polymorphic object of type
+ *      `std::variant<Arrangement_on_surface_2::Vertex_handle,
+ *                    Arrangement_on_surface_2::Halfedge_handle,
+ *                    Arrangement_on_surface_2::Face_handle>`.
```
Comment (albert-github) [inline Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arrangement_on_surface_2.h:1326]: In the nightly documentation test CGAL-6.0-Ic-67 (https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-6.0-Ic-67/index.html) we get for the 1.9.6 version of doxygen the warnings:
```
/home/cgal-testsuite/cgal_doc_build/CGAL-6.0-Ic-67/doc/Arrangement_on_surface_2/CGAL/Arr_vertical_decomposition_2.h:45: warning: explicit link request to 'variant' could not be resolved
/home/cgal-testsuite/cgal_doc_build/CGAL-6.0-Ic-67/doc/Arrangement_on_surface_2/CGAL/Arrangement_on_surface_2.h:1324: warning: explicit link request to 'variant' could not be resolved
```
(as of doxygen version 1.9.7 this problem has been solved).

As a workaround the problem can be solved to reformulate the:
```
 *      `std::variant<Arrangement_on_surface_2::Vertex_handle,
 *                    Arrangement_on_surface_2::Halfedge_handle,
 *                    Arrangement_on_surface_2::Face_handle>`.
 ```
to
```
 *      `std::variant<Arrangement_on_surface_2::Vertex_handle, Arrangement_on_surface_2::Halfedge_handle, Arrangement_on_surface_2::Face_handle>`.
 ```

analogous for doc/Arrangement_on_surface_2/CGAL/Arr_vertical_decomposition_2.h

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-84

**Pull request merged**